compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs

```diff
@@ -1378,14 +1378,21 @@ impl<'tcx> TypeErrCtxtExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
                         err.message =
                             vec![(rustc_errors::DiagnosticMessage::from(msg), Style::NoStyle)];
                     }
+                    let mut file = None;
                     err.span_label(
                         span,
                         format!(
                             "the trait `{}` is not implemented for `{}`",
                             old_pred.print_modifiers_and_trait_path(),
-                            old_pred.self_ty().skip_binder(),
+                            self.tcx.short_ty_string(old_pred.self_ty().skip_binder(), &mut file),
```

Comment:
Why introduce `short_ty_string` here? I'm not sure that the lifetime bounds we get with this change are really preferable.

Reply:
Although I agree that the prior output is likely "better", if we let the full type leak *anywhere* we have the real risk of ending up with a "wall of text" in *some* case. We could as a follow up change it so `short_ty_string` has the same output we had prior instead.

Reply:
But did you really handle all occurrences where types are printed in diagnostics in this PR? I would have expected many more changes in the diffs in the tests if that was the case. And if you haven't, I find this change kind of arbitrary.

Reply:
@b-naber I addressed all that I could find searching the codebase. There are many that we had already handled and this was an effort to find any stragglers.